crowbar_framework/app/models/updater_service.rb

```diff
@@ -55,15 +63,37 @@ def create_proposal
 
   def apply_role_pre_chef_call(old_role, role, all_nodes)
     @logger.debug("Updater apply_role_post_chef_call: entering #{all_nodes.inspect}")
-    # Remove [:updater][:one_shot_run] flag from node
+    # Remove [:updater][...] flags from node
     all_nodes.each do |n|
       node = NodeObject.find_node_by_name n
       unless node[:updater].nil?
-        node[:updater][:one_shot_run] = false
-        @logger.debug("Updater apply_role_post_chef_call: delete [:updater][:one_shot_run] for #{node.name} (#{node[:updater].inspect}")
+        if node.role? "updater"
+          node[:updater][:one_shot_run] = false
+          @logger.debug("Updater apply_role_post_chef_call: delete [:updater][:one_shot_run] for #{node.name} (#{node[:updater].inspect}")
+          node.save
+        end
+        if node.role? "updater-upgrade"
+        node[:updater][:upgrade_one_shot_run] = false
+        @logger.debug("Updater apply_role_post_chef_call: delete [:updater][:upgrade_one_shot_run] for #{node.name} (#{node[:updater].inspect}")
+        node.save
+        end
+      end
+    end
+    all_nodes.each do |n|
+      node = NodeObject.find_node_by_name n
+      if node.role? "updater-upgrade"
+        storage_role = role
+        # TODO: Make sure that storage_role.name does not already exist
+        storage_role.name= role.name+"-upgrade_storage"
+        storage_role.save
+        node.delete_from_run_list(role.name)
+        # TODO: Find out if there can be more than one role on the run_list -
+        # If so, handle that in a proper way
+        node.add_to_run_list("updater-upgrade", 2000)
```

Comment:
We don't really need a high priority here, do we? There should be just one role in this run list I assume